test/functional/sqlsrv/sqlsrv_ae_insert_sqltype_nchar_size.phpt

```diff
@@ -0,0 +1,270 @@
+--TEST--
+Test for inserting encrypted data of nchar types with different sizes
+--DESCRIPTION--
+Test implicit conversions between different nchar types of different sizes
+With Always Encrypted, implicit conversion works if:
+1. From input of SQLSRV_SQLTYPE_NCHAR(n) to a larger nchar(m) column where n <= m
+2. From input of SQLSRV_SQLTYPE_NCHAR(n) to a larger nvarchar(m) column where n <= m (m can be max)
+3. From input of SQLSRV_SQLTYPE_NVARCHAR(n) to a larger nchar(m) column where n <= m
+4. From input of SQLSRV_SQLTYPE_NVARCHAR(n) to a larger nvarchar(m) column where n <= m (m can be max)
+Without AlwaysEncrypted, implicit conversion between different binary types and sizes works
+--SKIPIF--
+<?php require('skipif_versions_old.inc'); ?>
+--FILE--
+<?php
+require_once('MsCommon.inc');
+
+$dataTypes = array("nchar", "nvarchar", "nvarchar(max)");
+$lengths = array(1, 8, 64, 512, 4000);
+$sqlTypes = array("SQLSRV_SQLTYPE_NCHAR", "SQLSRV_SQLTYPE_NVARCHAR", "SQLSRV_SQLTYPE_NVARCHAR('max')");
+$sqltypeLengths = $lengths;
+$inputValue = "d";
+
+$conn = AE\connect();
+foreach($dataTypes as $dataType) {
+    $maxcol = strpos($dataType, "(max)");
+    foreach($lengths as $m) {
+        if ($maxcol !== false) {
+            $typeFull = $dataType;
+        } else {
+            $typeFull = "$dataType($m)";
+        }
+        echo "\nTesting $typeFull:\n";
+            
+        // create table containing nchar(m) or nvarchar(m) columns
+        $tbname = "test_" . str_replace(array('(', ')'), '', $dataType) . $m;
+        $colMetaArr = array(new AE\ColumnMeta($typeFull, "c1", null, false));
+        AE\createTable($conn, $tbname, $colMetaArr);
```

Comment:
I don't mind having only one column in some tests, but at least put a comment to justify why only one column is needed